blog-service/2024-06-22-observability.md

```diff
@@ -0,0 +1,17 @@
+---
+title: Improved Data Handling for Window and Threshold-Based SLOs (Observability)
+image: https://help.sumologic.com/img/sumo-square.png
+keywords:
+  - slo
+  - observability
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+We are happy to introduce a new update for handling missing data in window and threshold-based Service Level Objectives (SLOs). Previously, any window without data was treated as a successful window and the threshold condition check was skipped. With this update, for window without data the threshold will be compared against **zero** to determine whether they are successful or not. Below are some of the key advantages of this change:
+
+* **Enhanced Accuracy**. Treating missing data as zero prevents incorrect assumptions about data quality, leading to more reliable SLO evaluations.
```

Comment:
Instead of these two points which are very generic, we can just give an example below.

`Consider an SLI definition where one expects a non-zero number of heartbeat log messages to be present in a given window. Previously such windows were treated as successful by default. Modelling such cases will become much easier with this change.`

Reply:
@apoorv-garg I guess the two advantages listed will explain "how modelling such cases will become easier". So, it okay to add example and those advantage points?

Reply:
it's not about accuracy, it's just a change in behaviour that will simplify achieving certain additional use cases. And consistency across windows is also not a customer goal. IMO the example is enough to drive home the value proposition.

Reply:
```
This change will enable expressing SLIs where missing data represents unsuccessful windows with ease. As an example, consider an SLI definition where one expects a non-zero number of heartbeat log messages to be present in a given window. Previously such windows were treated as successful by default. Modelling such cases will become much easier with this change.
```